src/cycles/eulerian.jl

```diff
@@ -0,0 +1,110 @@
+# Checking for existence of Eulerian circuit and Eulerian trail
+
+function isConnected(
+    graph::AbstractGraph{U},
+    ) where U<:Integer
+
+    visited = zeros(UInt8, nv(graph))
+
+    for v in vertices(graph)
+        if length(outneighbors(graph, v)) == 0
+            visited[v] = 2
+        end
+    end
+
+    for v in vertices(graph)
+
+        if length(outneighbors(graph, v)) == 0
+            continue
+        end
+        S = Vector{U}([v])
+        visited[v] = 1
+        while !isempty(S)
+            u = S[end]
+            w = 0
+            for n in outneighbors(graph, u)
+                if visited[n] == 0
+                    w = n
+                    break
+                end
+            end
+            if w != 0
+                visited[w] = 1
+                push!(S, w)
+            else
+                visited[u] = 2
+                pop!(S)
+            end
+        end
+
+        break
+    end
+
+    for v in vertices(graph)
+        if(visited[v]==0)
+            return false
+        end
+    end
+
+    return true
+end
+
+"""
```

Comment:
Maybe mention in this docstring, that a trail can have the same start and end vertex.